src/rustup-cli/rustup_mode.rs

```diff
@@ -687,7 +688,15 @@ fn toolchain_link(cfg: &Cfg, m: &ArgMatches) -> Result<()> {
     let ref toolchain = m.value_of("toolchain").expect("");
     let ref path = m.value_of("path").expect("");
     let toolchain = try!(cfg.get_toolchain(toolchain, true));
-
+    let mut pathbuf = PathBuf::from(path);
+
+    pathbuf.push("lib");
+    try!(utils::assert_is_directory(&pathbuf));
+    pathbuf.pop();
+    pathbuf.push("bin");
+    try!(utils::assert_is_directory(&pathbuf));
+    pathbuf.push("rustc");
+    try!(utils::assert_is_file(&pathbuf));
```

Comment:
I'm ok with adding this validation, but will you sink it into the `toolchain.install_from_dir` call below? This file is purely for argument parsing.